static/src/javascripts/projects/common/modules/navigation/search.js

```diff
@@ -0,0 +1,169 @@
+// @flow
+
+import bean from 'bean';
+import fastdom from 'fastdom';
+import $ from 'lib/$';
+import config from 'lib/config';
+import throttle from 'lodash/functions/throttle';
+
+const focusSearchField = (): void => {
+    const $input = $('input.gsc-input');
+    if ($input.length > 0) {
+        $input.focus();
+    }
+};
+
+class Search {
+    gcsUrl: string;
+    resultSetSize: number;
+
+    constructor(): void {
+        const toggle = document.querySelector('.js-search-toggle');
```

Comment:
Could we change the selector to use `getElementsByClassName('js-search-toggle')[0]` rather than `querySelector`? For a query on a classname  getElementsByClassName is faster/preferred.